app/models/health/qualifying_activity.rb

```diff
@@ -565,37 +565,39 @@ def compute_valid_unpayable?
     # :activity_outside_of_engagement_without_careplan -
     # :limit_months_without_careplan_exceeded -
     def compute_valid_unpayable
-      reasons = []
-      computed_procedure_valid = compute_procedure_valid?
-
-      # Only valid procedures can be valid unpayable
-      return nil unless computed_procedure_valid
-
-      # Unpayable if it is a valid procedure, but it didn't occur during an enrollment
-      reasons << :outside_enrollment if computed_procedure_valid && ! occurred_during_any_enrollment?
-
-      # Unpayable if this was a phone/video call where the client wasn't reached
-      reasons << :call_not_reached if reached_client == 'no' && ['phone_call', 'video_call'].include?(mode_of_contact)
-
-      # Signing a care plan is payable regardless of engagement status
-      return reasons if activity == 'pctp_signed'
-
-      # Outreach is limited by the outreach cut-off date, enrollment ranges, and frequency
-      if outreach?
-        reasons << :outreach_past_cutoff if date_of_activity > patient.outreach_cutoff_date
-        reasons << :outside_enrollment unless patient.contributed_dates.include?(date_of_activity)
-        reasons << :limit_outreaches_per_month_exceeded unless first_outreach_of_month_for_patient?
-        reasons << :limit_months_outreach_exceeded if number_of_outreach_activity_months > 3
-      else
-        # Non-outreach activities are payable at 1 per month before engagement unless there is a care-plan
-        unless patient_has_signed_careplan?
-          reasons << :limit_activities_per_month_without_careplan_exceeded unless first_non_outreach_of_month_for_patient?
-          reasons << :activity_outside_of_engagement_without_careplan if patient.engagement_date.blank? || date_of_activity > patient.engagement_date
-          reasons << :limit_months_without_careplan_exceeded if number_of_non_outreach_activity_months > 5
+      @compute_valid_unpayable ||= begin
```

Comment:
I don't think this is going to do what you expect with returns in it.  Maybe you need `yield`?

Reply:
Yes. `return` is incorrect... Rather than end up in dark corners of ruby, I replaced the `begin..end` with a private method.